src/ansi-c/library/unistd.c

```diff
@@ -385,3 +385,51 @@ __CPROVER_HIDE:;
   // Thus, modelling it as non-deterministic.
   return retval;
 }
+
+/* FUNCTION: syscall */
+
+#ifndef __CPROVER_ERRNO_H_INCLUDED
+#  include <errno.h>
+#  define __CPROVER_ERRNO_H_INCLUDED
+#endif
+
+long int __VERIFIER_nondet_long_int(void);
+int __VERIFIER_nondet_int(void);
+
+// This overapproximation is based on the syscall specification available at
+// https://man7.org/linux/man-pages/man2/syscall.2.html and
+// https://www.gnu.org/software/libc/manual/html_node/System-Calls.html.
+//
+// sysno is the system call number. The remaining arguments are the arguments
```

Comment:
Some system calls take pointer arguments and write data to the objects pointed to by these pointers.
To be overaproximating shouldn't this abstraction havoc all pointer arguments that were given to the `syscall` function ? This would maybe require handling this internally when typechecking calls to syscall.

Reply:
this lists all signatures for linux :
https://github.com/torvalds/linux/blob/master/include/linux/syscalls.h